assets/src/state.ts

```diff
@@ -66,10 +109,35 @@ export const reducer = (state: State, action: Action): State => {
     case "DESELECT_ROUTE":
       return {
         ...state,
-        selectedRouteIds: state.selectedRouteIds
-          .filter(id => id !== action.payload.routeId),
+        selectedRouteIds: state.selectedRouteIds.filter(
+          id => id !== action.payload.routeId
+        ),
+      }
+    case "SET_LOADING_TIMEPOINTS_FOR_ROUTE":
+      return {
+        ...state,
+        timepointsByRoute: {
+          ...state.timepointsByRoute,
+          [action.payload.routeId]: null,
+        },
+      }
+    case "SET_TIMEPOINTS_FOR_ROUTE":
+      return {
+        ...state,
+        timepointsByRoute: {
+          ...state.timepointsByRoute,
+          [action.payload.routeId]: action.payload.timepoints,
+        },
       }
     default:
       return state
   }
 }
+
+export type TimepointsForRouteId = (
+  routeId: RouteId
+) => LoadableTimepoints | undefined
+
+export const timepointsForRouteId = (
+  state: State
+): TimepointsForRouteId => routeId => state.timepointsByRoute[routeId]
```

Comment:
What's the difference between the names `timepointsByRoute`, `TimepointsByRouteId` and `[tT]imepointsForRouteId`? Could they all have the same name?

Is using a function to wrap the object so that we can get a real type on the function because of the weird typing restriction on objects?

Reply:
Fixed all this as discussed.

Reply:
Just so it's written down:
"timepoints for route" is for functions / actions etc that work on a single route.
"timepoints by route id" is the map of many route ids to their timepoints.
"timepoints for route id" no longer exists.
"timepoints by route" no longer exists.